python/pyspark/sql/functions.py

```diff
@@ -2206,6 +2207,10 @@ def pandas_udf(f=None, returnType=StringType()):
     |         8|      JOHN DOE|          22|
     +----------+--------------+------------+
     """
+    import pandas as pd
+    if isinstance(returnType, pd.Series):
+        returnType = from_pandas_dtypes(returnType)
```

Comment:
Is this necessary?

Reply:
I think we don't need this. Use consistent way to express the return type should be better.

Reply:
I agree having a consistent way to express return type is good.

The reason I added this is to enable this usage:

```
sample_df = df.filter(df.id == 1).toPandas()

def foo(df):
      ret = # Some transformation on the input pd.DataFrame
      return ret

foo_udf = pandas_udf(foo, foo(sample_df).dtypes)

df.groupBy('id').apply(foo_udf)
```

The pattern is quite useful in interactive usage. Here the user no longer needs to specify the return schema of the `foo` manually. And if the user changes the return columns of `foo`, they don't need to change the return type of `pandas_udf`. 

I am leaning towards keeping this but I am willing to be convinced.